dao/src/main/java/org/thingsboard/server/dao/device/DeviceProfileServiceImpl.java

```diff
@@ -122,15 +135,24 @@ public DeviceProfileInfo findDeviceProfileInfoById(TenantId tenantId, DeviceProf
     @Override
     public DeviceProfile saveDeviceProfile(DeviceProfile deviceProfile) {
         log.trace("Executing saveDeviceProfile [{}]", deviceProfile);
+        X509CertificateChainProvisionConfiguration x509Configuration = new X509CertificateChainProvisionConfiguration();
+        if (deviceProfile.getProfileData() != null && deviceProfile.getProfileData().getProvisionConfiguration() instanceof X509CertificateChainProvisionConfiguration) {
+            x509Configuration = (X509CertificateChainProvisionConfiguration) deviceProfile.getProfileData().getProvisionConfiguration();
+        }
+        if (x509Configuration.getCertificateValue() != null) {
```

Comment:
please move this `if statement` to the `if statement` above - this `if statement` will only work if previous `if statement` is **true**